### PR TITLE
Linting libraries phase 1

### PR DESCRIPTION
**Pull request by snebjorn**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/datorama/akita/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?

```
[x] Yes - updated typescript, I don't actually know if it's breaking anything
[ ] No
```

## Other information

- I had to update typescript to 3.3.4000 in order to fix a fatal error when running linting
  > FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

- I updated the workflow to run linting on the libraries as this is an on going effort #401, the rest will come later

- I tried to keep refactoring to a minimum, but some times the linting rules required a refactor - or my sanity 😉

- In some places I just gave up and added ignore comments 😭

### CODE REVIEW

File: libs/akita-ng-forms-manager/src/lib/forms-manager.query.ts [hunk 1]
```diff
@@ -1,8 +1,3 @@
 import { Query } from '@datorama/akita';
-import { FormsStore } from './forms-manager.store';
 
-export class FormsQuery<T> extends Query<T> {
-  constructor(store: FormsStore<T>) {
-    super(store);
-  }
-}
+export class FormsQuery<T> extends Query<T> {}
```
[the remark this answers is not in the document]
Comment (snebjorn) [reply libs/akita-ng-forms-manager/src/lib/forms-manager.query.ts:3]: it was deemed useless :)
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-useless-constructor.md

File: libs/akita/src/__tests__/persistStateInclude.spec.ts [hunk 2]
```diff
@@ -1,56 +1,33 @@
...
-    super();
-  }
-}
+class TodosStore extends EntityStore<any, any> {}
 
 @StoreConfig({
-  name: 'products'
+  name: 'products',
 })
-class ProductsStore extends EntityStore<any, any> {
```
[the remark this answers is not in the document]
Comment (snebjorn) [reply libs/akita/src/__tests__/persistStateInclude.spec.ts:19]: The constructor was deemed useless
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-useless-constructor.md

File: libs/akita-ng-router-store/src/lib/router.query.ts [hunk 3]
```diff
@@ -1,39 +1,45 @@
...
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class RouterQuery extends Query<RouterState> {
-  __navigationCancel = new Subject();
-  __navigationError = new Subject();
+  // TODO private?
+  __navigationCancel = new Subject<NavigationCancel>();
```
[the remark this answers is not in the document]
Comment (snebjorn) [reply libs/akita-ng-router-store/src/lib/router.query.ts:22]: the public accessors already exist

```ts
selectNavigationCancel(): Observable<NavigationCancel> {
  return this.__navigationCancel.asObservable();
}

selectNavigationError(): Observable<NavigationError> {
  return this.__navigationError.asObservable();
}
```

[the remark this answers is not in the document]
Comment (snebjorn) [reply libs/akita-ng-router-store/src/lib/router.query.ts:22]: oh wait, fire. Yep, adding

[the remark this answers is not in the document]
Comment (snebjorn) [reply libs/akita-ng-router-store/src/lib/router.query.ts:22]: we're treating `__vars` as internal and thus changing it isn't a breaking changes, yes?

[the remark this answers is not in the document]
Comment (snebjorn) [reply libs/akita-ng-router-store/src/lib/router.query.ts:22]:
```
nextNavigationCancel(event: NavigationCancel): void {
  return this.__navigationCancel.next(event);
}
```
or the angular flavor?  `emitNavigationCancel`

[the remark this answers is not in the document]
Comment (snebjorn) [reply libs/akita-ng-router-store/src/lib/router.query.ts:22]: since this is an angular plugin I'll go with the `emit` flavor

File: libs/akita-ng-forms-manager/src/lib/forms-manager.query.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (snebjorn) [reply libs/akita-ng-forms-manager/src/lib/forms-manager.query.ts:3]: it works as expected

[Playground Link](https://www.typescriptlang.org/play/index.html#code/PTAEEFQYwGwQwM4NAS2XUCAOBTKK4ZQAXAT11AHsAzUAKTgDc4BlKAJxS2KoCMArPDwDuACxRRRAWABQIVOhjC4pZBxxxiOACahGBUBiiUAdgmLsArlGKV2AOlAAVUTgQ5o8JG9ny4Nw1AYSh4YFABrD0oBIQQAGkMTXVEmDzgTVBNXTmJ0qA9zKxtLdhxfMAQUAFsUeHYSSiD0gHNLOGa3TGtRQ2QGZmAAYQBiYBZhFGpie1ly0AAJHFKAcnQMnAAPOCqsGA9YRAQALlmZA6RQADUcJLtQAG9ZUFATbZwjzAsUE2aAblPnsYzBZrLZ2AAKV5Vd6fTg-ACUDyezxI4gQ9ihHgAvC83v8ZM8AL4A0DNUo4YjgxGPAko0rEEoZABEixgwQSwhwMGM0IaoCZoAA1Ki0Bi8cjiTJJXNwBlvuY8vt0qBePtSpodHoDMRXC8cMJQJFSMI7NoEultHNSJRLNBlVBCERoTrKNo1rp-PkLlh2JRcOxiChOtRfVVUWU5GBooIbDMZLIgeZMKI-aAcSZ9Vcbto7OCmQBNDwAeRg2g8LBTWCZ8Pxicoe3swWa4IQlfsZJwFKpNdO8nzNrtGQQll45zWfBj0wWSxwq0C1EorrtAdAYgk0kjoBS6D0cE4FNIvVXXKIiECmJO8bOXmQADFF9onEVwqBNlokshrrd6vdpWdTEmIJQC+6aZverpPtY4R5gAIqkc4wXYKC2Ag1a1gB9Y4I2lDNkB4TtuSlLwjWoBAA)

File: libs/akita-ng-entity-service/src/lib/ng-entity-service.spec.ts [hunk 4]
```diff
@@ -1,33 +1,34 @@
...
 } from './setup';
-import { NgEntityServiceGlobalConfig, NG_ENTITY_SERVICE_CONFIG, defaultConfig } from './ng-entity-service.config';
-import { HttpMethod, NgEntityServiceNotifier } from './ng-entity-service-notifier';
-import { NgEntityServiceLoader } from './ng-entity-service.loader';
+import { HttpMethod } from './types';
 
 describe('NgEntityService', () => {
   describe('should merge config in order...', () => {
-    describe('', () => {
+    describe(undefined, () => {
```
Comment (NetanelBasal) [inline libs/akita-ng-entity-service/src/lib/ng-entity-service.spec.ts:20]: describe(undefined

???

File: libs/akita/src/lib/addEntities.ts [hunk 5]
```diff
@@ -1,46 +1,52 @@
...
+/** @internal */
+export function addEntities<S extends EntityState<E>, E>({ state, entities, idKey, options = {}, preAddEntity }: AddEntitiesParams<S, E>): { newState: S; newIds: any[] } | null {
+  const newEntities = {};
+  const newIds = [];
   let hasNewEntities = false;
 
-  for (const entity of entities) {
-    if (hasEntity(state.entities, entity[idKey]) === false) {
+  entities
+    .filter((entity) => hasEntity(state.entities, entity[idKey]) === false)
```
Comment (NetanelBasal) [inline libs/akita/src/lib/addEntities.ts:24]: You're looping the array twice. Please revert.

File: libs/akita/src/lib/arrayFind.ts [hunk 6]
```diff
@@ -95,7 +81,8 @@ export function arrayFind<T>(ids: ID, idKey?: string): (source: Observable<T[]>)
  */
 export function arrayFind<T>(ids: ID[], idKey?: string): (source: Observable<T[]>) => Observable<T[]>;
 export function arrayFind<T>(idsOrPredicate: ID[] | ID | ItemPredicate<T>, idKey?: string): (source: Observable<T[]>) => Observable<T[] | T> {
-  return function(source: Observable<T[]>) {
+  // TODO this looks like an OperatorFunction
```
Comment (NetanelBasal) [inline libs/akita/src/lib/arrayFind.ts:84]: It is

File: libs/akita/src/lib/entitiesToMap.ts [hunk 7]
```diff
@@ -15,29 +16,30 @@ export function entitiesToMap<S extends EntityState<E>, E>(state: S, options) {
...
+    for (let i = 0, { length } = ids; i < length; i += 1) {
       if (count === limitTo) break;
       const id = ids[i];
       const entity = entities[id];
-      const allPass = coerceArray(filterBy).every(fn => fn(entity, i));
+      const allPass = coerceArray(filterBy).every((fn) => fn(entity, i));
       if (allPass) {
         map[id] = entity;
-        count++;
+        count += 1;
```
Comment (NetanelBasal) [inline libs/akita/src/lib/entitiesToMap.ts:26]: why this change?

File: libs/akita/src/lib/entityStore.ts [hunk 8]
```diff
@@ -512,95 +534,106 @@ export class EntityStore<S extends EntityState = any, EntityType = getEntityType
    *
    * }
    */
-  createUIStore(initialState = {}, storeConfig: Partial<StoreConfigOptions> = {}) {
+  createUIStore(initialState = {}, storeConfig: Partial<StoreConfigOptions> = {}): EntityUIStore<any, EntityType> {
     const defaults: Partial<StoreConfigOptions> = { name: `UI/${this.storeName}`, idKey: this.idKey };
+    // TODO EntityUIStore uses EntityStore, and EntityStore uses EntityUIStore. This is bad, fix!
```
Comment (NetanelBasal) [inline libs/akita/src/lib/entityStore.ts:539]: ?

File: libs/akita/src/lib/query.ts [hunk 9]
```diff
@@ -1,22 +1,29 @@
...
 import { compareKeys } from './compareKeys';
+import { isDev } from './env';
+import { isFunction } from './isFunction';
+import { isString } from './isString';
+import { queryConfigKey, QueryConfigOptions } from './queryConfig';
+import { isBrowser } from './root';
+import { Store } from './store';
 import { ReturnTypes } from './types';
 
+/** @internal */
```
Comment (NetanelBasal) [inline libs/akita/src/lib/query.ts:12]: Why it's here?

File: libs/akita/src/lib/store.ts [hunk 10]
```diff
@@ -4,17 +4,23 @@ import { resetCustomAction, setAction } from './actions';
...
 import { isFunction } from './isFunction';
 import { isPlainObject } from './isPlainObject';
 import { isBrowser } from './root';
 import { configKey, StoreConfigOptions, UpdatableStoreConfigOptions } from './storeConfig';
-import { __stores__ } from './stores';
 import { commit, isTransactionInProcess } from './transaction';
 import { StoreCache, UpdateStateCallback } from './types';
 
+/** @internal */
+export const __stores__: { [storeName: string]: Store<any> } = {};
```
Comment (NetanelBasal) [inline libs/akita/src/lib/store.ts:18]: Wht it's here?

File: libs/akita/src/lib/transaction.ts [hunk 11]
```diff
@@ -86,16 +87,16 @@ export function applyTransaction<T>(action: () => T, thisArg = undefined): T {
  *
  */
 export function transaction() {
-  return function(target: any, propertyKey: string, descriptor: PropertyDescriptor) {
-    const originalMethod = descriptor.value;
+  return (target: any, propertyKey: string, descriptor: PropertyDescriptor): PropertyDescriptor => {
+    const descriptorCopy = { ...descriptor };
```
Comment (NetanelBasal) [inline libs/akita/src/lib/transaction.ts:91]: We don't really need a copy

Comment (snebjorn) [reply libs/akita/src/lib/transaction.ts:91]: I made a copy because of `descriptor.value = ...` that's mutating the input parameter
https://eslint.org/docs/rules/no-param-reassign

There are other ways to fix it. But mutation should be avoided

File: libs/akita/src/lib/store.ts (hunk 10 above)
Comment (snebjorn) [reply libs/akita/src/lib/store.ts:18]: I had to move it here to fix a cyclic import

File: libs/akita/src/lib/query.ts (hunk 9 above)
Comment (snebjorn) [reply libs/akita/src/lib/query.ts:12]: I had to move it here to fix a cyclic import

File: libs/akita/src/lib/entityStore.ts (hunk 8 above)
Comment (snebjorn) [reply libs/akita/src/lib/entityStore.ts:539]: The file have 2 classes defined and I was unable to move them into their own files as they're using each other. So I had to `/** eslint-disable max-classes-per-file */` and this TODO serves as a reminder to do something about it

File: libs/akita/src/lib/entitiesToMap.ts (hunk 7 above)
Comment (snebjorn) [reply libs/akita/src/lib/entitiesToMap.ts:26]: Ye I wasn't super happy with this either, but their reasoning seems sound
https://github.com/airbnb/javascript#variables--unary-increment-decrement

File: libs/akita-ng-entity-service/src/lib/ng-entity-service.spec.ts (hunk 4 above)
Comment (snebjorn) [reply libs/akita-ng-entity-service/src/lib/ng-entity-service.spec.ts:20]: having the same description is disallowed https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-identical-title.md

But I was using the describe blocks as a nameless grouping to get the setup hooks to setup correctly for each group.
Using undefined fixes the linting error

File: libs/akita/src/lib/persistState.ts [hunk 12]
```diff
@@ -235,21 +236,21 @@ export function persistState(params?: Partial<PersistStateParams>): PersistState
...
+    clear(): void {
       storage.clear();
     },
-    clearStore(storeName?: string) {
+    clearStore(storeName?: string): void {
       if (isNil(storeName)) {
-        const value = observify(storage.setItem(key, '{}'));
-        value.subscribe();
+        const value2 = observify(storage.setItem(key, '{}'));
+        value2.subscribe();
```
Comment (snebjorn) [inline libs/akita/src/lib/persistState.ts:253]: Please suggest a better name

File: libs/akita/src/lib/persistState.ts [hunk 13]
```diff
@@ -258,8 +259,8 @@ export function persistState(params?: Partial<PersistStateParams>): PersistState
 
         if (storageState[storeName]) {
           delete storageState[storeName];
-          const value = observify(storage.setItem(key, serialize(storageState)));
-          value.subscribe();
+          const value2 = observify(storage.setItem(key, serialize(storageState)));
+          value2.subscribe();
```
Comment (snebjorn) [inline libs/akita/src/lib/persistState.ts:263]: Please suggest a better name

File: libs/akita/src/lib/plugins/persistForm/persistNgFormPlugin.ts [hunk 14]
```diff
@@ -61,12 +66,12 @@ export class PersistNgFormPlugin<T = any> extends AkitaPlugin {
...
+      Object.keys(this.initialValue).forEach((stateKey) => {
+        const value2 = this.initialValue[stateKey];
+        if (Array.isArray(value2) && this.builder) {
           const formArray = this.form.controls[stateKey];
           this.cleanArray(formArray);
-          value.forEach((v, i) => {
+          value2.forEach((v, i) => {
             this.form.get(stateKey).insert(i, (this.params.arrControlFactory as Function)(v));
           });
         }
```
Comment (snebjorn) [inline libs/akita/src/lib/plugins/persistForm/persistNgFormPlugin.ts:77]: Please suggest a better name

File: libs/akita/src/__tests__/dirtyCheck.spec.ts [hunk 15]
```diff
@@ -137,102 +149,102 @@ describe('DirtyCheck', () => {
       });
 
       describe('dirtyPath', () => {
-        const widgetsStore = new WidgetsStore({ some: { nested: { value: '' } } });
-        const widgetsQuery = new WidgetsQuery(widgetsStore);
-        const dirtyCheck = new DirtyCheckPlugin(widgetsQuery);
-        it('should not return dirty for isPathDirty', function() {
+        const widgetsStore2 = new WidgetsStore({ some: { nested: { value: '' } } });
+        const widgetsQuery2 = new WidgetsQuery(widgetsStore2);
+        const dirtyCheck2 = new DirtyCheckPlugin(widgetsQuery2);
```
Comment (snebjorn) [inline libs/akita/src/__tests__/dirtyCheck.spec.ts:154]: Please suggest a better name

File: libs/akita/src/__tests__/dirtyCheck.spec.ts [hunk 16]
```diff
@@ -511,151 +527,142 @@ describe('DirtyCheckEntity', () => {
...
   describe('Passing ids', () => {
-    let widgetsStore = new WidgetsStore();
-    let widgetsQuery = new WidgetsQuery(widgetsStore);
+    const widgetsStore2 = new WidgetsStore();
+    const widgetsQuery2 = new WidgetsQuery(widgetsStore2);
     _id = 0;
-    let collection = new EntityDirtyCheckPlugin(widgetsQuery, { entityIds: [1, 2] });
-    widgetsStore.add([createWidget(), createWidget(), createWidget()]);
-    collection.setHead([1, 2]);
+    let collection2 = new EntityDirtyCheckPlugin(widgetsQuery2, { entityIds: [1, 2] });
```
Comment (snebjorn) [inline libs/akita/src/__tests__/dirtyCheck.spec.ts:533]: Please suggest a better name

File: libs/akita/src/__tests__/entityStore.spec.ts [hunk 17]
```diff
@@ -454,7 +454,7 @@ describe('EntitiesStore', () => {
 });
 
 describe('Custom ID', () => {
-  let store2 = new TodosStoreCustomID();
+  const store2 = new TodosStoreCustomID();
```
Comment (snebjorn) [inline libs/akita/src/__tests__/entityStore.spec.ts:457]: Please suggest a better name

File: libs/akita/src/__tests__/pagination.spec.ts [hunk 18]
```diff
@@ -314,7 +314,7 @@ describe('cacheTimeout', () => {
   });
 });
 
-let store3 = new TodosStore();
+const store3 = new TodosStore();
```
Comment (snebjorn) [inline libs/akita/src/__tests__/pagination.spec.ts:317]: Please suggest a better name

File: libs/akita/src/__tests__/pagination.spec.ts [hunk 19]
```diff
@@ -264,15 +264,15 @@ describe('Paginator', () => {
...
       paginator.clearCache({ clearStore: false });
-      let lengthAfterCacheClear = query.getAll().length;
+      const lengthAfterCacheClear = query.getAll().length;
       expect(initialLength).toEqual(lengthAfterCacheClear);
     });
   });
 });
 
-let store2 = new TodosStore();
+const store2 = new TodosStore();
```
Comment (snebjorn) [inline libs/akita/src/__tests__/pagination.spec.ts:275]: Please suggest a better name

File: libs/akita/src/__tests__/stateHistory.spec.ts [hunk 20]
```diff
@@ -106,177 +106,179 @@ describe('StateHistory', () => {
...
 
-const store2 = new CounterStore();
-const query2 = new CounterQuery(store2);
-const stateHistory2 = new StateHistoryPlugin(query2, { maxAge: 1 });
-
 describe('StateHistory - Limit', () => {
-  store2._setState(state => {
+  const store = new CounterStore();
+  const query = new CounterQuery(store);
+  const stateHistory2 = new StateHistoryPlugin(query, { maxAge: 1 });
```
Comment (snebjorn) [inline libs/akita/src/__tests__/stateHistory.spec.ts:222]: This was called `stateHistory2` before but please suggest a better name

File: libs/akita/src/__tests__/store.spec.ts [hunk 21]
```diff
@@ -9,29 +9,33 @@ interface State {
...
+  });
+
   it('should initialized the store', () => {
     expect(store._value()).toEqual(state);
   });
 
   it('should select slice from the store', () => {
     const spy = jest.fn();
-    store._select(state => state.theme).subscribe(spy);
+    store._select((state2) => state2.theme).subscribe(spy);
```
Comment (snebjorn) [inline libs/akita/src/__tests__/store.spec.ts:38]: Please suggest a better name

File: libs/akita/src/__tests__/store.spec.ts [hunk 22]
```diff
@@ -46,13 +50,13 @@ describe('Store', () => {
...
   it('should set a new state', () => {
     const spy = jest.fn();
-    store._select(state => state.theme).subscribe(spy);
-    store._setState(state => {
+    store._select((state2) => state2.theme).subscribe(spy);
+    store._setState((state2) => {
       return {
-        ...state,
+        ...state2,
         theme: {
```
Comment (snebjorn) [inline libs/akita/src/__tests__/store.spec.ts:57]: Please suggest a better name

File: libs/akita/src/lib/queryEntity.ts [hunk 23]
```diff
@@ -302,14 +321,16 @@ export class QueryEntity<S extends EntityState, EntityType = getEntityType<S>, I
...
+
   selectEntityAction(action?: EntityActions): Observable<IDType[] | EntityAction<IDType>> {
     if (isUndefined(action)) {
       return this.store.selectEntityAction$;
     }
     return this.store.selectEntityAction$.pipe(
-      filter(ac => ac.type === action),
-      map(action => action.ids)
+      filter((ac) => ac.type === action),
+      map((action2) => action2.ids)
```
Comment (snebjorn) [inline libs/akita/src/lib/queryEntity.ts:333]: Please suggest a better name

File: libs/akita/src/__tests__/queryEntity.spec.ts [hunk 24]
```diff
@@ -965,40 +982,44 @@ describe('Sort by - Query Level', () => {
...
+        res = _res;
+      });
 
     expect(res[0].price).toEqual(40);
     expect(res[1].price).toEqual(10);
     expect(res[2].price).toEqual(3);
   });
 
-  it('should let selectAll win', () => {
+  it('should let selectAll win (2)', () => {
```
Comment (snebjorn) [inline libs/akita/src/__tests__/queryEntity.spec.ts:1004]: Please suggest a better description

File: libs/akita/src/__tests__/queryEntity.spec.ts [hunk 25]
```diff
@@ -1072,37 +1092,36 @@ describe('selectAll - limit to and filterBy and sorting', () => {
...
       })
-      .subscribe(_res => {
+      .subscribe((_res) => {
         res = _res;
       });
     expect(res.length).toBe(5);
   });
 
-  it('should support array', () => {
+  it('should support array (2)', () => {
```
Comment (snebjorn) [inline libs/akita/src/__tests__/queryEntity.spec.ts:1116]: Please suggest a better description

File: libs/akita/src/__tests__/removeEntities.spec.ts [hunk 26]
```diff
@@ -62,7 +62,7 @@ describe('removeEntities', () => {
     store.remove();
   });
 
-  it('should reset the actives when removed', () => {
+  it('should reset the actives when removed (2)', () => {
```
Comment (snebjorn) [inline libs/akita/src/__tests__/removeEntities.spec.ts:65]: Please suggest a better description

File: libs/akita-ng-forms-manager/src/lib/forms-manager.spec.ts [hunk 27]
```diff
@@ -681,11 +682,11 @@ describe('FormsManager', () => {
       errors: null,
       touched: false,
       pristine: true,
-      pending: false
+      pending: false,
     });
   }));
 
-  it('should update the store with forms value', fakeAsync(() => {
+  it('should update the store with forms value (2)', fakeAsync(() => {
```
Comment (snebjorn) [inline libs/akita-ng-forms-manager/src/lib/forms-manager.spec.ts:689]: Please suggest a better description

File: libs/akita-ng-forms-manager/src/lib/forms-manager.spec.ts [hunk 28]
```diff
@@ -1295,14 +1296,14 @@ describe('FormsManager', () => {
...
-      }
+            pending: false,
+          },
+        },
+      },
     });
   }));
 
-  it('should listen to changes - control', fakeAsync(() => {
+  it('should listen to changes - control (2)', fakeAsync(() => {
```
Comment (snebjorn) [inline libs/akita-ng-forms-manager/src/lib/forms-manager.spec.ts:1306]: Please suggest a better description

**Comment by snebjorn** [comment]
Just noticed the CI is silently failing :(

> (node:2968) UnhandledPromiseRejectionWarning: Error: Nx failed to execute runtimeCacheInputs defined in nx.json failed:
> Command failed: type .eslintrc
>     at Hasher.<anonymous> (/home/runner/work/akita/akita/node_modules/@nrwl/workspace/src/tasks-runner/hasher.js:59:27)
>     at Generator.throw (<anonymous>)
>     at rejected (/home/runner/work/akita/akita/node_modules/@nrwl/workspace/src/tasks-runner/hasher.js:6:65)
>     at processTicksAndRejections (internal/process/task_queues.js:97:5)
> (node:2968) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 12)
> (node:2968) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

File: libs/akita/src/lib/entitiesToMap.ts (hunk 7 above)
Comment (snebjorn) [reply libs/akita/src/lib/entitiesToMap.ts:26]: I'm going to revert this rule, as it's quite annoying and I believe TS+prettier will guard against the pit falls this rule tries to prevent

Comment (NetanelBasal) [reply libs/akita/src/lib/entitiesToMap.ts:26]: Great

File: libs/akita/src/lib/persistState.ts (hunk 12 above)
Comment (NetanelBasal) [reply libs/akita/src/lib/persistState.ts:253]: saved$

File: libs/akita/src/lib/persistState.ts (hunk 13 above)
Comment (NetanelBasal) [reply libs/akita/src/lib/persistState.ts:263]: afterSaved$

File: libs/akita/src/lib/plugins/persistForm/persistNgFormPlugin.ts (hunk 14 above)
Comment (NetanelBasal) [reply libs/akita/src/lib/plugins/persistForm/persistNgFormPlugin.ts:77]: currentValue

File: libs/akita/src/lib/queryEntity.ts (hunk 23 above)
Comment (NetanelBasal) [reply libs/akita/src/lib/queryEntity.ts:333]: Use destructuring.

File: libs/akita-ng-entity-service/src/lib/ng-entity-service.spec.ts (hunk 4 above)
Comment (NetanelBasal) [reply libs/akita-ng-entity-service/src/lib/ng-entity-service.spec.ts:20]: Disable this rule or give it a name, please. `undefined` is not acceptable.

**Comment by snebjorn** [comment]
@NetanelBasal any idea why the CI stopped running on this PR?

**Comment by NetanelBasal** [comment]
No
